### PR TITLE
fix potential panic when read last line

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <chen.two.cs@gmail.com>

Close issue: https://github.com/pingcap/sysutil/issues/29

This PR does 2 things:
1. fix potential panic when read log last line.
2. add recover to avoid process exit cause by search log panic.

### CODE REVIEW

**Comment by crazycs520** [comment]
@baurine @lonng PTAL

File: service.go [hunk 1]
```diff
@@ -35,7 +38,17 @@ func NewDiagnosticsServer(logFile string) *DiagnosticsServer {
...
-func (d *DiagnosticsServer) SearchLog(req *pb.SearchLogRequest, stream pb.Diagnostics_SearchLogServer) error {
+func (d *DiagnosticsServer) SearchLog(req *pb.SearchLogRequest, stream pb.Diagnostics_SearchLogServer) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			buf := make([]byte, 4096)
+			stackSize := runtime.Stack(buf, false)
+			buf = buf[:stackSize]
+			err = fmt.Errorf(fmt.Sprintf("search log panic, %v, stack is %v", r, string(buf)))
+			log.Error(err.Error())
+		}
```
Comment (breeswish) [inline service.go:49]: Should we return `err` to the outside?

**Review by breeswish — COMMENTED** [review]
The rest LGTM. Remember to upgrade the sysutil for branches.

File: service.go (hunk 1 above)
Comment (crazycs520) [reply service.go:49]: The error already returns to the outside since line 47 assign value to `err`.

**Review by breeswish — APPROVED** [review]

**Pull request merged**